src/btree/bt_page.c

```diff
@@ -52,16 +49,13 @@ __evict_force_check(WT_SESSION_IMPL *session, WT_PAGE *page, uint32_t flags)
 	if (page->modify == NULL)
 		return (0);
 
-	/*
-	 * If the page was recently split in-memory, don't force it out: we
-	 * hope eviction will find it first.
-	 */
-	if (!__wt_txn_visible_all(session, page->modify->first_dirty_txn))
-		return (0);
-
 	/* Trigger eviction on the next page release. */
 	__wt_page_evict_soon(page);
 
+	/* If eviction cannot succeed, don't try. */
+	if (!__wt_page_can_evict(session, page, 1))
+		return (0);
+
```

Comment:
@michaelcahill, why are we setting `WT_READGEN_OLDEST` on a page that cannot be evicted? I guess the question is how expensive is `__wt_page_can_evict`, are we willing to keep calling it it on pages that cannot be evicted, each time the page is released?